Add deploy recipe for gae

idosh opened the pull request:
Heavily based on https://github.com/ebidel/polymer-gmail/blob/master/app.yaml

Fixes #634

robdodson commented on deploy.sh:
```diff
@@ -0,0 +1,39 @@
+
+DEPLOY_VERSION=$1
+
+if [ -z "$GAE_PROJECT" ]
```
why not specify the project name in app.yaml?

idosh replied:
I'm a conservative guy, I don't like to hardcode names.

robdodson commented on deploy.sh:
```diff
@@ -0,0 +1,39 @@
+
+# Build it.
+echo "Building $DEPLOY_VERSION"
+#gulp
```
What's the gulp comment referencing?

idosh replied:
by mistake, I commented for testing but forgot to uncomment

robdodson commented on deploy.sh:
```diff
@@ -0,0 +1,39 @@
+# Build it.
+echo "Building $DEPLOY_VERSION"
+#gulp
+cp app.yaml dist/app.yaml
```
Hm this is a neat trick. I usually have the dist directory part of my app.yaml, but I kind of like this approach

robdodson commented:
Hey @idosh thanks for working on this! Would you mind creating a directory for these files in `https://github.com/PolymerElements/polymer-starter-kit/tree/master/docs` that's where all the recipes live. If you feel like doing a write up on the steps that would also be :trophy:

idosh commented:
Hi @robdodson,
I've added a new recipe in docs and squashed the commits.
Waiting for your review...

On Thu, Jan 7, 2016, 21:05 Rob Dodson <notifications@github.com> wrote:

> Hey @idosh <https://github.com/idosh> thanks for working on this! Would
> you mind creating a directory for these files in
> https://github.com/PolymerElements/polymer-starter-kit/tree/master/docs
> that's where all the recipes live. If you feel like doing a write up on the
> steps that would also be [image: :trophy:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PolymerElements/polymer-starter-kit/pull/640#issuecomment-169775589>
> .
>

robdodson commented:
LGTM. @samccone or @ebidel, either of you want to take a second look?

ebidel commented on docs/deploy-to-google-app-engine.md:
```diff
@@ -0,0 +1,109 @@
+        # Build it.
+        echo "Building $DEPLOY_VERSION"
+        gulp
+        cp app.yaml dist/app.yaml
```
the gulp file could do this

idosh replied:
Are you referring to the whole deploy script or just the copying of the
yaml file?

On Fri, Jan 8, 2016, 20:22 Eric Bidelman <notifications@github.com> wrote:

> In docs/deploy-to-google-app-engine.md
> <https://github.com/PolymerElements/polymer-starter-kit/pull/640#discussion_r49219366>
> :
>
> > +        #!/usr/bin/env bash
> > +
> > +        GAE_PROJECT=psk
> > +        DEPLOY_VERSION=$1
> > +
> > +        if [ -z "$DEPLOY_VERSION" ]
> > +        then
> > +          TAG=`git describe`
> > +          # GAE doesn't allow periods
> > +          DEPLOY_VERSION=${TAG//.}
> > +        fi
> > +
> > +        # Build it.
> > +        echo "Building $DEPLOY_VERSION"
> > +        gulp
> > +        cp app.yaml dist/app.yaml
>
> the gulp file could do this
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PolymerElements/polymer-starter-kit/pull/640/files#r49219366>
> .
>

robdodson replied:
I think Eric's referring specifically to the copy line

idosh replied:
IMHO it would be easier to add two files (app.yaml & deploy.sh) than modifying the gulpfile and add the files

robdodson replied:
hm if the app.yaml file was already in the `app/` dir I think the gulp copy task would just pick it up automatically: https://github.com/PolymerElements/polymer-starter-kit/blob/master/gulpfile.js#L147-L157

@idosh could you give this a shot and see if it works?

ebidel commented on docs/deploy-to-google-app-engine.md:
```diff
@@ -0,0 +1,109 @@
+
+1.  Run the deploy script
+
+        ./deploy.sh
```
this needs a version arg

idosh replied:
Not necessarily, as I wrote, if a deploy version is not provided the last
git tag will be used.

On Fri, Jan 8, 2016, 20:22 Eric Bidelman <notifications@github.com> wrote:

> In docs/deploy-to-google-app-engine.md
> <https://github.com/PolymerElements/polymer-starter-kit/pull/640#discussion_r49219415>
> :
>
> > +        gulp
> > +        cp app.yaml dist/app.yaml
> > +
> > +        echo "Deploying $DEPLOY_VERSION"
> > +        gcloud preview app deploy dist/app.yaml --project $GAE_PROJECT --promote --version $DEPLOY_VERSION
> > +
> > +    You have to set `GAE_PROJECT` variable to your GAE project id.
> > +    A deploy version can be provided as a parameter for the script, if not provided the latest git tag will be used.
> > +
> > +1.  Add execution permission to the script
> > +
> > +        chmod +x deploy.sh
> > +
> > +1.  Run the deploy script
> > +
> > +        ./deploy.sh
>
> this needs a version arg
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PolymerElements/polymer-starter-kit/pull/640/files#r49219415>
> .
>

robdodson replied:
If the user hasn't tagged it yet will the script fail? It may be good to show an example with the version arg anyway just so folks realize it's an option

idosh replied:
done :)

On Tue, Jan 12, 2016 at 5:06 PM Rob Dodson <notifications@github.com> wrote:

> In docs/deploy-to-google-app-engine.md
> <https://github.com/PolymerElements/polymer-starter-kit/pull/640#discussion_r49464700>
> :
>
> > +        gulp
> > +        cp app.yaml dist/app.yaml
> > +
> > +        echo "Deploying $DEPLOY_VERSION"
> > +        gcloud preview app deploy dist/app.yaml --project $GAE_PROJECT --promote --version $DEPLOY_VERSION
> > +
> > +    You have to set `GAE_PROJECT` variable to your GAE project id.
> > +    A deploy version can be provided as a parameter for the script, if not provided the latest git tag will be used.
> > +
> > +1.  Add execution permission to the script
> > +
> > +        chmod +x deploy.sh
> > +
> > +1.  Run the deploy script
> > +
> > +        ./deploy.sh
>
> If the user hasn't tagged it yet will the script fail? It may be good to
> show an example with the version arg anyway just so folks realize it's an
> option
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PolymerElements/polymer-starter-kit/pull/640/files#r49464700>
> .
>

robdodson commented on docs/deploy-to-google-app-engine.md:
```diff
@@ -0,0 +1,109 @@
+
+1.  Create a new GCP project using the [Developers Console](https://console.developers.google.com/home/dashboard)
+
+1.  Add app.yaml to your app root folder
```
"Add app.yaml to your `app` directory"

idosh replied:
I mean "Add app.yaml to your project root folder".
I don't think placing it in the app folder is a good choice, what do you think @robdodson ?

robdodson replied:
What downside do you see to putting it in the app directory?

idosh replied:
The app folder is more about code & styling whereas the root folder contains the tools and configurations

robdodson replied:
Hm I guess that's valid. One reason that I was thinking it might be nice in the current app directory is I believe, as it is currently written, it may allow you to use the local app engine dev server if it's in there.

`cd app && dev_appserver.py .` but I haven't tested it yet :)

I'm fine with it living in the root directory though. Would you mind updating that sentence in the PR to just say 'Add app.yaml to your project root' ?

idosh replied:
done

robdodson commented:
Thanks @idosh for all the work on this! :100:

idosh commented:
@robdodson much appreciated!
